### PR TITLE
Initial project scaffolding for dev tools pages

### PR DESCRIPTION
**Pull request by LogvinovLeon**
## Checklist:

*   [ ] Prefix PR title with `[WIP]` if necessary.
*   [ ] Prefix PR title with bracketed package name(s) corresponding to the changed package(s). For example: `[sol-cov] Fixed bug`.
*   [ ] Add tests to cover changes as needed.
*   [ ] Update documentation as needed.
*   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: packages/dev-tools-pages/README.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+## Website & 0x Portal DApp
```
Comment (fabioberger) [inline packages/dev-tools-pages/README.md]: Change this.

File: packages/dev-tools-pages/README.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+```bash
+yarn lint
+```
+
+### Resources
+
+##### Toolkit
+
+*   [BassCSS toolkit](http://basscss.com/)
```
Comment (fabioberger) [inline packages/dev-tools-pages/README.md]: Replace BassCSS with `styled-components`: https://www.styled-components.com. We actually want to move away from BassCSS and instead use https://rebassjs.org/, so we could add that and remove BassCSS.

File: packages/dev-tools-pages/package.json [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    "description": "0x Dev tools pages",
+    "scripts": {
+        "build": "node --max_old_space_size=8192 ../../node_modules/.bin/webpack --mode production",
+        "build:dev": "../../node_modules/.bin/webpack --mode development",
+        "clean": "shx rm -f public/bundle*",
+        "lint": "tslint --project . 'ts/**/*.ts' 'ts/**/*.tsx'",
+        "dev": "webpack-dev-server --mode development --content-base public",
+        "deploy_dogfood": "npm run build; aws s3 sync ./public/. s3://dogfood.0xproject.com --profile 0xproject --region us-east-1 --grants read=uri=http://acs.amazonaws.com/groups/global/AllUsers",
+        "deploy_staging": "npm run build; aws s3 sync ./public/. s3://staging-0xproject --profile 0xproject --region us-east-1 --grants read=uri=http://acs.amazonaws.com/groups/global/AllUsers",
+        "deploy_live": "DEPLOY_ROLLBAR_SOURCEMAPS=true npm run build; aws s3 sync ./public/. s3://0xproject.com --profile 0xproject --region us-east-1 --grants read=uri=http://acs.amazonaws.com/groups/global/AllUsers --exclude *.map.js"
```
Comment (fabioberger) [inline packages/dev-tools-pages/package.json]: Remove deploy commands for now

File: packages/dev-tools-pages/package.json [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    },
+    "author": "Fabio Berger",
```
Comment (fabioberger) [inline packages/dev-tools-pages/package.json]: Remove.

File: packages/dev-tools-pages/package.json [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    "license": "Apache-2.0",
+    "dependencies": {
+        "basscss": "^8.0.3",
+        "bowser": "^1.9.3",
```
Comment (fabioberger) [inline packages/dev-tools-pages/package.json]: What is using this? Remove.

File: packages/dev-tools-pages/package.json [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+        "less": "^2.7.2",
+        "lodash": "^4.17.5",
+        "query-string": "^6.0.0",
+        "react": "^16.4.2",
+        "react-copy-to-clipboard": "^5.0.0",
```
Comment (fabioberger) [inline packages/dev-tools-pages/package.json]: Remove.

File: packages/dev-tools-pages/package.json [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+        "deploy_live": "DEPLOY_ROLLBAR_SOURCEMAPS=true npm run build; aws s3 sync ./public/. s3://0xproject.com --profile 0xproject --region us-east-1 --grants read=uri=http://acs.amazonaws.com/groups/global/AllUsers --exclude *.map.js"
+    },
+    "author": "Fabio Berger",
+    "license": "Apache-2.0",
+    "dependencies": {
+        "basscss": "^8.0.3",
+        "bowser": "^1.9.3",
+        "less": "^2.7.2",
+        "lodash": "^4.17.5",
+        "query-string": "^6.0.0",
```
Comment (fabioberger) [inline packages/dev-tools-pages/package.json]: Remove.

File: packages/dev-tools-pages/package.json [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+        "react": "^16.4.2",
+        "react-copy-to-clipboard": "^5.0.0",
+        "react-document-title": "^2.0.3",
+        "react-dom": "^16.4.2",
+        "react-helmet": "^5.2.0",
```
Comment (fabioberger) [inline packages/dev-tools-pages/package.json]: ?

File: packages/dev-tools-pages/package.json [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+        "styled-components": "^3.3.0",
+        "thenby": "^1.2.3"
```
Comment (fabioberger) [inline packages/dev-tools-pages/package.json]: ?

File: packages/dev-tools-pages/package.json [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+    },
+    "devDependencies": {
+        "@types/lodash": "4.14.104",
+        "@types/node": "*",
+        "@types/numeral": "^0.0.22",
```
Comment (fabioberger) [inline packages/dev-tools-pages/package.json]: ?

File: packages/dev-tools-pages/public/index.html [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+    <meta property="og:title" content="0x" />
+    <meta property="og:description" content="" />
+    <meta property="og:image" content="/images/og_image.png" />
+    <title>0x: The Protocol for Trading Tokens</title>
+    <link rel="icon" type="image/png" href="/images/favicon/favicon-2-32x32.png" sizes="32x32" />
+    <link rel="icon" type="image/png" href="/images/favicon/favicon-2-16x16.png" sizes="16x16" />
+    <link rel="stylesheet" href="/css/basscss_responsive_custom.css">
+    <link rel="stylesheet" href="/css/basscss_responsive_padding.css">
+    <link rel="stylesheet" href="/css/basscss_responsive_margin.css">
+    <link rel="stylesheet" href="/css/basscss_responsive_type_scale.css">
```
Comment (fabioberger) [inline packages/dev-tools-pages/public/index.html]: Remove.

File: packages/dev-tools-pages/public/index.html [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+</head>
+
+<body style="margin: 0px; min-width: 355px;">
+    <!-- Heap SDK -->
+    <script type="text/javascript">
+        window.heap = window.heap || [], heap.load = function (e, t) { window.heap.appid = e, window.heap.config = t = t || {}; var r = t.forceSSL || "https:" === document.location.protocol, a = document.createElement("script"); a.type = "text/javascript", a.async = !0, a.src = (r ? "https:" : "http:") + "//cdn.heapanalytics.com/js/heap-" + e + ".js"; var n = document.getElementsByTagName("script")[0]; n.parentNode.insertBefore(a, n); for (var o = function (e) { return function () { heap.push([e].concat(Array.prototype.slice.call(arguments, 0))) } }, p = ["addEventProperties", "addUserProperties", "clearEventProperties", "identify", "resetIdentity", "removeEventProperty", "setEventProperties", "track", "unsetEventProperty"], c = 0; c < p.length; c++)heap[p[c]] = o(p[c]) };
+        heap.load("410099666");
```
Comment (fabioberger) [inline packages/dev-tools-pages/public/index.html]: Remove.

File: packages/dev-tools-pages/public/index.html [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+                fjs = d.getElementsByTagName(s)[0];
+            if (d.getElementById(id)) return;
+            js = d.createElement(s);
+            js.id = id;
+            js.src = '//connect.facebook.net/en_US/sdk.js#xfbml=1&version=v2.8&appId=1687545238205192';
+            fjs.parentNode.insertBefore(js, fjs);
+        })(document, 'script', 'facebook-jssdk');
+    </script>
+    <div id="app"></div>
+    <!-- End Facebook SDK -->
```
Comment (fabioberger) [inline packages/dev-tools-pages/public/index.html]: Remove FB.

File: packages/dev-tools-pages/public/index.html [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+    <script>
+        (function (h, o, t, j, a, r) {
+            h.hj = h.hj || function () { (h.hj.q = h.hj.q || []).push(arguments) };
+            h._hjSettings = { hjid: 935597, hjsv: 6 };
+            a = o.getElementsByTagName('head')[0];
+            r = o.createElement('script'); r.async = 1;
+            r.src = t + h._hjSettings.hjid + j + h._hjSettings.hjsv;
+            a.appendChild(r);
+        })(window, document, 'https://static.hotjar.com/c/hotjar-', '.js?sv=');
+    </script>
```
Comment (fabioberger) [inline packages/dev-tools-pages/public/index.html]: Remove.

File: packages/dev-tools-pages/ts/globals.d.ts [hunk 15]
```diff
@@ -0,0 +1,10 @@
+declare module 'whatwg-fetch';
+declare module 'react-document-title';
+declare module 'react-ga';
```
Comment (fabioberger) [inline packages/dev-tools-pages/ts/globals.d.ts]: Remove.

File: packages/dev-tools-pages/ts/index.tsx [hunk 16]
```diff
@@ -0,0 +1,18 @@
+import * as React from 'react';
+import { render } from 'react-dom';
+import { MetaTags } from 'ts/components/meta_tags';
+import { Landing } from 'ts/pages/landing/landing';
+
+import 'basscss/css/basscss.css';
+import 'less/all.less';
```
Comment (fabioberger) [inline packages/dev-tools-pages/ts/index.tsx]: Remove both.

File: packages/dev-tools-pages/ts/index.tsx [hunk 17]
```diff
@@ -0,0 +1,18 @@
...
+
+const DOCUMENT_TITLE = '0x: The Protocol for Trading Tokens';
+const DOCUMENT_DESCRIPTION = 'An Open Protocol For Decentralized Exchange On The Ethereum Blockchain';
```
Comment (fabioberger) [inline packages/dev-tools-pages/ts/index.tsx]: Update.

File: packages/dev-tools-pages/ts/utils/configs.ts [hunk 18]
```diff
@@ -0,0 +1,12 @@
+const BASE_URL = window.location.origin;
+
+export const configs = {
+    BACKEND_BASE_PROD_URL: 'https://website-api.0xproject.com',
+    BACKEND_BASE_STAGING_URL: 'https://staging-website-api.0xproject.com',
+    BASE_URL,
+    DOMAIN_STAGING: 'staging-0xproject.s3-website-us-east-1.amazonaws.com',
+    DOMAIN_DOGFOOD: 'dogfood.0xproject.com',
+    DOMAINS_DEVELOPMENT: ['0xproject.localhost:3572', 'localhost:3572', '127.0.0.1'],
+    DOMAIN_PRODUCTION: '0xproject.com',
```
Comment (fabioberger) [inline packages/dev-tools-pages/ts/utils/configs.ts]: Remove add these.

File: packages/dev-tools-pages/ts/utils/configs.ts [hunk 19]
```diff
@@ -0,0 +1,12 @@
...
+    GOOGLE_ANALYTICS_ID: 'UA-98720122-1',
```
Comment (fabioberger) [inline packages/dev-tools-pages/ts/utils/configs.ts]: Remove.

File: packages/dev-tools-pages/ts/utils/utils.ts [hunk 20]
```diff
@@ -0,0 +1,46 @@
...
+        return _.includes(configs.DOMAINS_DEVELOPMENT, window.location.host);
+    },
+    isStaging(): boolean {
+        return _.includes(window.location.href, configs.DOMAIN_STAGING);
+    },
+    isDogfood(): boolean {
+        return _.includes(window.location.href, configs.DOMAIN_DOGFOOD);
+    },
+    isProduction(): boolean {
+        return _.includes(window.location.href, configs.DOMAIN_PRODUCTION);
```
Comment (fabioberger) [inline packages/dev-tools-pages/ts/utils/utils.ts]: Remove.

File: packages/dev-tools-pages/webpack.config.js [hunk 21]
```diff
@@ -0,0 +1,92 @@
...
+    let plugins = [];
+    if (argv.mode === 'development') {
+        config.mode = 'development';
+    } else {
+        config.mode = 'production';
+        plugins = plugins.concat([
+            new webpack.DefinePlugin({
+                'process.env': {
+                    NODE_ENV: JSON.stringify(process.env.NODE_ENV),
+                    GIT_SHA: JSON.stringify(GIT_SHA),
```
Comment (fabioberger) [inline packages/dev-tools-pages/webpack.config.js]: Remove.

File: packages/dev-tools-pages/package.json (hunk 8 above)
Comment (LogvinovLeon) [reply packages/dev-tools-pages/package.json]: We use it. It's for head section metatags

File: packages/dev-tools-pages/ts/index.tsx (hunk 16 above)
Comment (LogvinovLeon) [reply packages/dev-tools-pages/ts/index.tsx]: Why? Wouldn't they write CSS in that file?

File: packages/dev-tools-pages/README.md [hunk 22]
```diff
@@ -0,0 +1,58 @@
...
+
+```bash
+yarn lint
+```
+
+### Resources
+
+##### Toolkit
+
+*   [Styled Components](https://www.styled-components.com/)
```
Comment (fabioberger) [inline packages/dev-tools-pages/README.md]: Let's add `rebass` here.

File: packages/dev-tools-pages/package.json [hunk 23]
```diff
@@ -0,0 +1,55 @@
...
+        "@types/react-helmet": "^5.0.6",
+        "@types/react-router-dom": "^4.0.4",
+        "@types/react-tap-event-plugin": "0.0.30",
+        "awesome-typescript-loader": "^5.2.1",
+        "copyfiles": "^2.0.0",
+        "css-loader": "0.23.x",
+        "less-loader": "^4.1.0",
+        "make-promises-safe": "^1.1.0",
+        "raw-loader": "^0.5.1",
+        "rollbar-sourcemap-webpack-plugin": "^2.4.0",
```
Comment (fabioberger) [inline packages/dev-tools-pages/package.json]: remove.

File: packages/dev-tools-pages/ts/index.tsx (hunk 16 above)
Comment (fabioberger) [reply packages/dev-tools-pages/ts/index.tsx]: No. We don't want any CSS or LESS files anymore. All styling should be in the components themselves.

File: packages/dev-tools-pages/webpack.config.js [hunk 24]
```diff
@@ -0,0 +1,86 @@
...
+                loader: 'awesome-typescript-loader',
+            },
+            {
+                test: /\.md$/,
+                use: 'raw-loader',
+            },
+            {
+                test: /\.less$/,
+                loader: 'style-loader!css-loader!less-loader',
+                exclude: /node_modules/,
```
Comment (fabioberger) [inline packages/dev-tools-pages/webpack.config.js]: Remove.

File: packages/dev-tools-pages/webpack.config.js [hunk 25]
```diff
@@ -0,0 +1,86 @@
...
+            },
+            {
+                test: /\.css$/,
+                loaders: ['style-loader', 'css-loader'],
```
Comment (fabioberger) [inline packages/dev-tools-pages/webpack.config.js]: Remove.

**Pull request merged**